google/cloud/bigquery/v2/minimal/internal/dataset_metadata.cc

```diff
@@ -0,0 +1,74 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Implementation of internal interface for Bigquery V2 Dataset resource.
+
+#include "google/cloud/bigquery/v2/minimal/internal/dataset_metadata.h"
+#include "google/cloud/common_options.h"
+#include "google/cloud/internal/absl_str_cat_quiet.h"
+#include "google/cloud/internal/absl_str_join_quiet.h"
+#include "google/cloud/internal/api_client_header.h"
+#include <memory>
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace bigquery_v2_minimal_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+DatasetMetadata::DatasetMetadata(std::shared_ptr<DatasetRestStub> child)
+    : child_(std::move(child)),
+      api_client_header_(
+          google::cloud::internal::ApiClientHeader("bigquery_v2_dataset")) {}
+
+StatusOr<GetDatasetResponse> DatasetMetadata::GetDataset(
+    rest_internal::RestContext& context, GetDatasetRequest const& request) {
+  SetMetadata(context);
```

Comment:
I am still surprised there is no specific set of request parameters associated with these requests, but we can always fix that later and let you make progress.